data/talkactions/scripts/create_item.lua

```diff
@@ -8,10 +8,17 @@ function onSay(player, words, param)
 	end
 
 	local split = param:split(",")
+	if not split[1] then
+		player:sendCancelMessage("You need to specify an item name or id.")
+		return false
+	end
 
 	local itemType = ItemType(split[1])
 	if itemType:getId() == 0 then
-		itemType = ItemType(tonumber(split[1]))
+		if tonumber(split[1]) then
+			itemType = ItemType(tonumber(split[1]))
+		end
+
 		if itemType:getId() == 0 then
```

Comment:
I think you misunderstood me or I misunderstood you. On a second read I can't relate your comment to the code.